server/service/endpoint_utils.go

```diff
@@ -131,6 +134,40 @@ func makeDecoder(iface interface{}) kithttp.DecodeRequestFunc {
 			if jsonExpected && nilBody {
 				return nil, errors.New("Expected JSON Body")
 			}
+
+			queryTagValue, ok := f.Tag.Lookup("query")
```

Comment:
This part and below is to add support for a new tag query to specify query parameters as part of the request struct as you'll see in the `listSoftwareStruct`.